### PR TITLE
Split guest hot plugging

### PR DESCRIPTION
**Pull request by tbaev**
Separate module hotplugging.md in smaller tests for easier debugging.

This PR will **brake** OSD test on new product, more information in progress ticket. poo#96230

- Related ticket: https://progress.opensuse.org/issues/96230
- Needles: N / A
- Verification run: void [KVM](http://openqa.qam.suse.cz/tests/35136#dependencies) void [XEN](http://openqa.qam.suse.cz/tests/35117#dependencies)

### CODE REVIEW

**Comment by waynechen55** [comment]
- I went through this new test flow. It seems that the hotplugging test itself does not have new code or modified code. It is just split further into sub-modules to make it look nice. 

- But this new test flow can still have repercussion for the openQA automation test, because you firstly split then integrate all back into one test suite, so you need thorough testing. Not only on 15-SP4, but also cover all test scenarios happen on openQA, including those in milestone group. IMHO.

**Review by grisu48 — COMMENTED** [review]
LGTM so far. I cannot see any issues with the VR and this PR is only about splitting `hotplugging.pm` into submodules, so, I believe it's sane.

One thing: In `main_common.pm` we would have the chance to add custom triggers to disable specific hotplugging tests by adding settings. Perhaps we should add

    loadtest 'virtualization/universal/hotplugging_memory' if (get_var('VIRT_HOTPLUGGING_MEMORY', '1'));

WDYT?

**Comment by tbaev** [comment]
@waynechen55  
You are correct, the idea is only to split the test for ease of readability and debugging.

I wasn't aware of the  'cleanup' module as we don't use it on Void.

I have looked deeper into the problem and run some tests, it seems you are absolutely correct that the guest doesn't have time to fully boot. 

I have added `wait_guest_online` at the start of the modules to resolve this.

And started two new VR for [KVM](https://openqa.suse.de/tests/7958403#) and [XEN](https://openqa.suse.de/tests/7958404#)

File: tests/virt_autotest/hotplugging_utils.pm [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# SUSE's openQA tests
+#
+# Copyright 2020 SUSE LLC
```
Comment (b10n1k) [inline tests/virt_autotest/hotplugging_utils.pm:3]: As a new module the date can be ommitted. see _CONTRIBUTING.md_

File: tests/virt_autotest/hotplugging_utils.pm [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+use warnings;
+use utils;
+use version_utils;
+use testapi;
+use File::Basename;
+use Utils::Architectures;
+
+our @EXPORT = qw(set_vcpus set_guest_memory get_guest_memory get_disk_image_name reset_guest try_attach);
+
+# Subroutine to change number of vCPUs for guest
```
Comment (b10n1k) [inline tests/virt_autotest/hotplugging_utils.pm:21]: Why not proper perldoc here?

File: tests/virt_autotest/hotplugging_utils.pm (hunk 1 above)
Comment (tbaev) [reply tests/virt_autotest/hotplugging_utils.pm:3]: I will remove the year.

File: tests/virt_autotest/hotplugging_utils.pm (hunk 2 above)
Comment (tbaev) [reply tests/virt_autotest/hotplugging_utils.pm:21]: It is a small set of utils mainly self explanatory, by looking at other utils my team is using I decided to make it based on them and not use perldoc format.

**Comment by waynechen55** [comment]
> @waynechen55 You are correct, the idea is only to split the test for ease of readability and debugging.
> 
> I wasn't aware of the 'cleanup' module as we don't use it on Void.
> 
> I have looked deeper into the problem and run some tests, it seems you are absolutely correct that the guest doesn't have time to fully boot.
> 
> I have added `wait_guest_online` at the start of the modules to resolve this.
> 
> And started two new VR for [KVM](https://openqa.suse.de/tests/7958403#) and [XEN](https://openqa.suse.de/tests/7958404#)

@tbaev  

I listed out all test suites on osd that need to be covered by your verification runs. Please help validate one by one. And if there is failure happens to any of them, please also help make sure the failure is caused by this commit:
[gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403)
[gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881)
[gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361)
[gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362)
[gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363)
[gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364)
[gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365)
[gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366)
[gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317)
[gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071)
[uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048)
[uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment.
[uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014)
[uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment.
[uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282)
[uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017)
[uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018)
[uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment.
[uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020)
[uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment.

**Comment by tbaev** [comment]
**I will update results from VRs here. WIP**

 [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403)
Result: Passed [VR](https://openqa.suse.de/tests/7964973#)
 [gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881)
Result: Running [VR](https://openqa.suse.de/tests/7965023#live)
 [gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361)
Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964977#) 
 [gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362)
Result: Passed [VR](https://openqa.suse.de/tests/7964978#)
 [gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363)
Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964980#)
 [gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364)
Result: Passed [VR](https://openqa.suse.de/tests/7964983#)
 [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365)
Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964984)
 [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366)
Result: Passed [VR](https://openqa.suse.de/tests/7964985)
 [gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317)
Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964986)
 [gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071)
Result: Running [VR](https://openqa.suse.de/tests/7964987)
 [uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048)
Result: Passed [VR](https://openqa.suse.de/tests/7965000)
 [uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment.
Result: Skip
 [uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014)
Result: Passed [VR](https://openqa.suse.de/tests/7965001)
 [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment.
Result: Skip
 [uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282)
Result: Running [VR](https://openqa.suse.de/tests/7965002#)
 [uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017)
Result: Running [VR](https://openqa.suse.de/tests/7965012#)
 [uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018)
Result: Running [VR](https://openqa.suse.de/tests/7965020#)
 [uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment.
Result: Skip
 [uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020)
Result: Running [VR](https://openqa.suse.de/tests/7965024#)
 [uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment.
Result: Skip

So far I am observing two issues :
1 Softfailure at hotplugging_network_interfaces that can be found in tests before the test split. I will ignore the soft failure and mark it as "passed".
2 Failure at sriov_network_card_pci_passthrough , this module is run before hot plugging and I am thinking to retry the tests where we have this error. I don't think sriov_network_card_pci_passthrough is related to code in this commit.

**Comment by waynechen55** [comment]
> 

Soft failure is acceptable. And please help re-run those failed at SR-IOV. Thanks.

**Comment by waynechen55** [comment]
> @waynechen55 **I will update results from VRs here. WIP**
> 
> [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403) Result: Passed [VR](https://openqa.suse.de/tests/7964973#) [gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881) Result: Running [VR](https://openqa.suse.de/tests/7965023#live) [gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964977#) [gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362) Result: Passed [VR](https://openqa.suse.de/tests/7964978#) [gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964980#) [gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364) Result: Passed [VR](https://openqa.suse.de/tests/7964983#) [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964984) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) Result: Passed [VR](https://openqa.suse.de/tests/7964985) [gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964986) [gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071) Result: Running [VR](https://openqa.suse.de/tests/7964987) [uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048) Result: Passed [VR](https://openqa.suse.de/tests/7965000) [uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014) Result: Passed [VR](https://openqa.suse.de/tests/7965001) [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282) Result: Running [VR](https://openqa.suse.de/tests/7965002#) [uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017) Result: Running [VR](https://openqa.suse.de/tests/7965012#) [uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018) Result: Running [VR](https://openqa.suse.de/tests/7965020#) [uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020) Result: Running [VR](https://openqa.suse.de/tests/7965024#) [uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment. Result: Skip
> 
> So far I am observing two issues : 1 Softfailure at hotplugging_network_interfaces that can be found in tests before the test split. I will ignore the soft failure and mark it as "passed". 2 Failure at sriov_network_card_pci_passthrough , this module is run before hot plugging and I am thinking to retry the tests where we have this error. I don't think sriov_network_card_pci_passthrough is related to code in this commit.

Soft failure is acceptable. And please help re-run those failed at SR-IOV. Thanks.

**Comment by tbaev** [comment]
> > @waynechen55 **I will update results from VRs here. WIP**
> > [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403) Result: Passed [VR](https://openqa.suse.de/tests/7964973#) [gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881) Result: Running [VR](https://openqa.suse.de/tests/7965023#live) [gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964977#) [gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362) Result: Passed [VR](https://openqa.suse.de/tests/7964978#) [gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964980#) [gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364) Result: Passed [VR](https://openqa.suse.de/tests/7964983#) [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964984) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) Result: Passed [VR](https://openqa.suse.de/tests/7964985) [gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964986) [gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071) Result: Running [VR](https://openqa.suse.de/tests/7964987) [uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048) Result: Passed [VR](https://openqa.suse.de/tests/7965000) [uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014) Result: Passed [VR](https://openqa.suse.de/tests/7965001) [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282) Result: Running [VR](https://openqa.suse.de/tests/7965002#) [uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017) Result: Running [VR](https://openqa.suse.de/tests/7965012#) [uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018) Result: Running [VR](https://openqa.suse.de/tests/7965020#) [uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020) Result: Running [VR](https://openqa.suse.de/tests/7965024#) [uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment. Result: Skip
> > So far I am observing two issues : 1 Softfailure at hotplugging_network_interfaces that can be found in tests before the test split. I will ignore the soft failure and mark it as "passed". 2 Failure at sriov_network_card_pci_passthrough , this module is run before hot plugging and I am thinking to retry the tests where we have this error. I don't think sriov_network_card_pci_passthrough is related to code in this commit.
> 
> * Soft failure is acceptable. The test itself will not fail and terminate at soft failure. Eventually the whole test flow will be executed.
> * And please help re-run those failed at SR-IOV. Thanks.

I have re-run the failing SR-IOV tests, and all of them Passed. I think we are safe to merge this change.

**Comment by waynechen55** [comment]
> > > @waynechen55 **I will update results from VRs here. WIP**
> > > [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403) Result: Passed [VR](https://openqa.suse.de/tests/7964973#) [gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881) Result: Running [VR](https://openqa.suse.de/tests/7965023#live) [gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964977#) [gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362) Result: Passed [VR](https://openqa.suse.de/tests/7964978#) [gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964980#) [gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364) Result: Passed [VR](https://openqa.suse.de/tests/7964983#) [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964984) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) Result: Passed [VR](https://openqa.suse.de/tests/7964985) [gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964986) [gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071) Result: Running [VR](https://openqa.suse.de/tests/7964987) [uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048) Result: Passed [VR](https://openqa.suse.de/tests/7965000) [uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014) Result: Passed [VR](https://openqa.suse.de/tests/7965001) [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282) Result: Running [VR](https://openqa.suse.de/tests/7965002#) [uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017) Result: Running [VR](https://openqa.suse.de/tests/7965012#) [uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018) Result: Running [VR](https://openqa.suse.de/tests/7965020#) [uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020) Result: Running [VR](https://openqa.suse.de/tests/7965024#) [uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment. Result: Skip
> > > So far I am observing two issues : 1 Softfailure at hotplugging_network_interfaces that can be found in tests before the test split. I will ignore the soft failure and mark it as "passed". 2 Failure at sriov_network_card_pci_passthrough , this module is run before hot plugging and I am thinking to retry the tests where we have this error. I don't think sriov_network_card_pci_passthrough is related to code in this commit.
> > 
> > 
> > 
> > * Soft failure is acceptable. The test itself will not fail and terminate at soft failure. Eventually the whole test flow will be executed.
> > * And please help re-run those failed at SR-IOV. Thanks.
> 
> I have re-run the failing SR-IOV tests, and all of them Passed. I think we are safe to merge this change.

I did not see your verification runs for the following two:
[gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365)
[gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366)
The hotplugging test in these two still use the all-in-one hotplugging instead of split ones.

**Comment by tbaev** [comment]
> > > > @waynechen55 **I will update results from VRs here. WIP**
> > > > [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403) Result: Passed [VR](https://openqa.suse.de/tests/7964973#) [gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881) Result: Running [VR](https://openqa.suse.de/tests/7965023#live) [gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964977#) [gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362) Result: Passed [VR](https://openqa.suse.de/tests/7964978#) [gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964980#) [gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364) Result: Passed [VR](https://openqa.suse.de/tests/7964983#) [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964984) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) Result: Passed [VR](https://openqa.suse.de/tests/7964985) [gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964986) [gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071) Result: Running [VR](https://openqa.suse.de/tests/7964987) [uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048) Result: Passed [VR](https://openqa.suse.de/tests/7965000) [uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014) Result: Passed [VR](https://openqa.suse.de/tests/7965001) [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282) Result: Running [VR](https://openqa.suse.de/tests/7965002#) [uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017) Result: Running [VR](https://openqa.suse.de/tests/7965012#) [uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018) Result: Running [VR](https://openqa.suse.de/tests/7965020#) [uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020) Result: Running [VR](https://openqa.suse.de/tests/7965024#) [uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment. Result: Skip
> > > > So far I am observing two issues : 1 Softfailure at hotplugging_network_interfaces that can be found in tests before the test split. I will ignore the soft failure and mark it as "passed". 2 Failure at sriov_network_card_pci_passthrough , this module is run before hot plugging and I am thinking to retry the tests where we have this error. I don't think sriov_network_card_pci_passthrough is related to code in this commit.
> > > 
> > > 
> > > 
> > > * Soft failure is acceptable. The test itself will not fail and terminate at soft failure. Eventually the whole test flow will be executed.
> > > * And please help re-run those failed at SR-IOV. Thanks.
> > 
> > 
> > I have re-run the failing SR-IOV tests, and all of them Passed. I think we are safe to merge this change.
> 
> I did not see your verification runs for the following two: [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) The hotplugging test in these two still use the all-in-one hotplugging instead of split ones.

There are in my post where I keep track of the runs, here are the links again
 [KVM](https://openqa.suse.de/tests/7968238#) and the [XEN](https://openqa.suse.de/tests/7964985) run

**Comment by tbaev** [comment]
> > > > > > @waynechen55 **I will update results from VRs here. WIP**
> > > > > > [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7958403) Result: Passed [VR](https://openqa.suse.de/tests/7964973#) [gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7960881) Result: Running [VR](https://openqa.suse.de/tests/7965023#live) [gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924361) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964977#) [gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924362) Result: Passed [VR](https://openqa.suse.de/tests/7964978#) [gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924363) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964980#) [gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924364) Result: Passed [VR](https://openqa.suse.de/tests/7964983#) [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964984) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) Result: Passed [VR](https://openqa.suse.de/tests/7964985) [gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7924317) Result: Failed at sriov_network_card_pci_passthrough [VR](https://openqa.suse.de/tests/7964986) [gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7939071) Result: Running [VR](https://openqa.suse.de/tests/7964987) [uefi-gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/7924048) Result: Passed [VR](https://openqa.suse.de/tests/7965000) [uefi-gi-guest_developing-on-host_developing-xen](https://openqa.suse.de/tests/7924049) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles12sp5-kvm](https://openqa.suse.de/tests/7924014) Result: Passed [VR](https://openqa.suse.de/tests/7965001) [uefi-gi-guest_developing-on-host_sles12sp5-xen](https://openqa.suse.de/tests/7924015) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_developing-on-host_sles15sp3-kvm](https://openqa.suse.de/tests/7927282) Result: Running [VR](https://openqa.suse.de/tests/7965002#) [uefi-gi-guest_developing-on-host_sles15sp3-xen](https://openqa.suse.de/tests/7924017) Result: Running [VR](https://openqa.suse.de/tests/7965012#) [uefi-gi-guest_sles12sp5-on-host_developing-kvm](https://openqa.suse.de/tests/7924018) Result: Running [VR](https://openqa.suse.de/tests/7965020#) [uefi-gi-guest_sles12sp5-on-host_developing-xen](https://openqa.suse.de/tests/7924019) This test is blocked by bug. You can skip it at the moment. Result: Skip [uefi-gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924020) Result: Running [VR](https://openqa.suse.de/tests/7965024#) [uefi-gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924021) This test is blocked by bug. You can skip it at the moment. Result: Skip
> > > > > > So far I am observing two issues : 1 Softfailure at hotplugging_network_interfaces that can be found in tests before the test split. I will ignore the soft failure and mark it as "passed". 2 Failure at sriov_network_card_pci_passthrough , this module is run before hot plugging and I am thinking to retry the tests where we have this error. I don't think sriov_network_card_pci_passthrough is related to code in this commit.
> > > > > 
> > > > > 
> > > > > 
> > > > > * Soft failure is acceptable. The test itself will not fail and terminate at soft failure. Eventually the whole test flow will be executed.
> > > > > * And please help re-run those failed at SR-IOV. Thanks.
> > > > 
> > > > 
> > > > I have re-run the failing SR-IOV tests, and all of them Passed. I think we are safe to merge this change.
> > > 
> > > 
> > > I did not see your verification runs for the following two: [gi-guest_sles15sp3-on-host_developing-kvm](https://openqa.suse.de/tests/7924365) [gi-guest_sles15sp3-on-host_developing-xen](https://openqa.suse.de/tests/7924366) The hotplugging test in these two still use the all-in-one hotplugging instead of split ones.
> > 
> > 
> > There are in my post where I keep track of the runs, here are the links again [KVM](https://openqa.suse.de/tests/7968238#) and the [XEN](https://openqa.suse.de/tests/7964985) run
> 
> The links are for 15sp4 on 15sp3 kvm/xen. I am talking about 15sp3 on 15sp4 kvm/xen. You did not verify 15sp3 on 15sp4 kvm/xen.

My mistake. Here are the two missing runs. They "Passed" too.

https://openqa.suse.de/tests/7972442

https://openqa.suse.de/tests/7972443

**Comment by waynechen55** [comment]
LGTM. Have you done any verification run for MU test ?

**Comment by tbaev** [comment]
> LGTM. Have you done any verification run for MU test ?

Yes, [KVM](http://openqa.qam.suse.cz/tests/35136#dependencies) and [XEN](http://openqa.qam.suse.cz/tests/35117#dependencies)

**Comment by waynechen55** [comment]
> > LGTM. Have you done any verification run for MU test ?
> 
> Yes, [KVM](http://openqa.qam.suse.cz/tests/35136#dependencies) and [XEN](http://openqa.qam.suse.cz/tests/35117#dependencies)

LGTM

**Review by grisu48 — APPROVED** [review]
LGTM!

**Comment by tonyyuan1** [comment]
The PR looks good to me!

**Pull request merged**